features/config/upgrade/src/main/java/org/opennms/config/upgrade/UpgradeConfigService.java

```diff
@@ -45,16 +46,22 @@
 
     private final ConfigurationManagerService cm;
     private final DataSource dataSource;
+    private final boolean skipConfigUpgrades;
 
     @Inject
     public UpgradeConfigService(final ConfigurationManagerService cm,
-                                final DataSource dataSource) {
+                                final DataSource dataSource,
+                                @Value( "${skipConfigUpgrades:false}" )
+                                final boolean skipConfigUpgrades) {
         this.cm = cm;
         this.dataSource = dataSource;
+        this.skipConfigUpgrades = skipConfigUpgrades;
     }
 
     @Override
     public void afterPropertiesSet() throws Exception {
-        new LiquibaseUpgrader(cm).runChangelog("changelog-cm/changelog-cm.xml", dataSource.getConnection());
+        if (!skipConfigUpgrades) { // TODO: Patrick discuss with Jesse: ok?
```

Comment:
@j-white I need a way to disable the upgrade for Integration tests. This is what I came up with. Is that the proper way to do that or do you have a better / more elegant idea? Thanks!

Reply:
Tricky. I think this approach is fine.